cloudbuild.yaml

```diff
@@ -1,9 +1,9 @@
 # See https://cloud.google.com/cloud-build/docs/build-config
-timeout: 1800s
+timeout: 2400s
 options:
   substitution_option: ALLOW_LOOSE
 steps:
-  - name: 'gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20220609-2e4c91eb7e'
+  - name: 'gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20230516-623029c2c0'
```

Comment:
I see the latest one right now was released on the 22nd: [`v20230522-312425ae46`](https://console.cloud.google.com/gcr/images/k8s-staging-test-infra/global/gcb-docker-gcloud@sha256:88e6bbc4bb2154947de91bf3ec6df92c1d4a6f1b65612f84b4b8b20c1a88f9a6/details?tab=info). Should we use that?